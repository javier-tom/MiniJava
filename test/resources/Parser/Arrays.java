class Main {
    public static void main(String[] args) {
        System.out.println((new Thing()).run());
    }
}

class Thing {
    int[] arr;
    public int run() {
        int[] a;
        int i;

        i = 0;
        a = new int[10];
        arr = new int [10];
        a[0] = 0;

        while (i < a.length) {
            a[i] = i;
            i = i + 1;
        }

        i = i - 1;
        while (0-1 < i) {
            arr[i] = a[i];
            i = i - 1;
        }
        return i;
    }
}