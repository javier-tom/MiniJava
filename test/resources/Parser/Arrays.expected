Program
  MainClass Main (line 1)
    Print  (line 3)
      Call run (line 3)
        on class:
          NewObject Thing (line 3)
        parameters:

  Class Thing (line 7)
    fields:
      VarDecl int[] arr (line 8)
    MethodDecl run (line 9)
      returns int
      params:
      variables:
        VarDecl int[] a (line 10)
        VarDecl int i (line 11)
      Assign to i (line 13)
        new value:
          0 (line 13)
      Assign to a (line 14)
        new value:
          NewArray (line 14)
            size:
              10 (line 14)
      Assign to arr (line 15)
        new value:
          NewArray (line 15)
            size:
              10 (line 15)
      ArrayAssign to a (line 16)
        index:
          0 (line 16)
        value:
          0 (line 16)
      While (line 18)
        condition:
          LessThan (line 18)
            i (line 18)
            10 (line 18)
        ArrayAssign to a (line 19)
          index:
            i (line 19)
          value:
            i (line 19)
        Assign to i (line 20)
          new value:
            Plus (line 20)
              i (line 20)
              1 (line 20)
      Assign to i (line 23)
        new value:
          Minus (line 23)
            i (line 23)
            1 (line 23)
      While (line 24)
        condition:
          Minus (line 24)
            0 (line 24)
            LessThan (line 24)
              1 (line 24)
              i (line 24)
        ArrayAssign to arr (line 25)
          index:
            i (line 25)
          value:
            ArrayLookup (line 25)
              on:
                a (line 25)
              at index:
                i (line 25)
        Assign to i (line 26)
          new value:
            Minus (line 26)
              i (line 26)
              1 (line 26)
      Return
        i (line 28)